### ISSUE TITLE
preview_definition not working anymore

### ISSUE THREAD

**Issue by smichaud** [glepnir/lspsaga.nvim#217]
I am pretty sure this feature (preview_definition) worked when I first installed lspsaga, but now I get an error every time I call it:

![image](https://user-images.githubusercontent.com/2299396/123484234-047dc100-d5d6-11eb-8b26-efc6b4dbae11.png)
 
Other feature (including finder) are still working.

**Comment by Pasqat** [comment]
Yeah I encountered this issue some weeks ago...I don't know what is wrong...

**Comment by johnybx** [comment]
Seems that this patch fixes the issue:
```diff
diff --git a/lua/lspsaga/provider.lua b/lua/lspsaga/provider.lua
index 22c0d86..3bc7321 100644
--- a/lua/lspsaga/provider.lua
+++ b/lua/lspsaga/provider.lua
@@ -443,7 +443,15 @@ function lspfinder.preview_definition(timeout_ms)
     print("No location found: " .. method)
     return nil
   end
-  result = {vim.tbl_deep_extend("force", {}, unpack(result))}
+
+  -- Result is not empty - there must be at least one element at this point
+  for k, v in pairs(result) do
+      if v ~= nil then
+        non_empty_res = v
+      end
+  end
+
+  result = {vim.tbl_deep_extend("force", {}, non_empty_res)}
 
   if vim.tbl_islist(result) and not vim.tbl_isempty(result[1]) then
     local uri = result[1].result[1].uri or result[1].result[1].targetUri
```
at least for me ( tested lua, python, php language servers ). Main problem is that the result in table is expected at index 1 but for example for pyright the result is returned at index 2 in table while [unpack](http://www.lua.org/manual/5.1/manual.html#pdf-unpack) by default index from 1. I need to read up little bit documentation for nvim lsp and mainly `vim.lsp.buf_request_sync` than I could create pull request. I am not sure if this is also related to this issue https://github.com/glepnir/lspsaga.nvim/issues/230.

**Comment by johnybx** [comment]
seems that there is already waiting merge request https://github.com/glepnir/lspsaga.nvim/pull/196 for this.
